ffi/build.rs

```diff
@@ -41,10 +59,61 @@ fn main() {
                 }
             }
         }
-    } else {
-        fs::remove_file("mitmproxy_rs/windows-redirector.exe").ok();
-        for wd_file in windivert_files {
-            fs::remove_file(format!("mitmproxy_rs/{wd_file}")).ok();
+    }
+
+    #[cfg(target_os = "macos")]
+    {
+        // macos-certificate-truster binary
+        copy_dir(
+            Path::new("../macos-certificate-truster/macos-certificate-truster.app/"),
+            Path::new("mitmproxy_rs/macos-certificate-truster.app"),
+        );
+        if fs::create_dir_all("mitmproxy_rs/macos-certificate-truster.app/Contents/MacOS/").is_ok() && cfg!(debug_assertions) {
+            fs::copy(
+                "../target/debug/macos-certificate-truster",
+                "mitmproxy_rs/macos-certificate-truster.app/Contents/MacOS/macos-certificate-truster",
+            )
+        } else {
+            fs::copy(
+                "../target/release/macos-certificate-truster",
+                "mitmproxy_rs/macos-certificate-truster.app/Contents/MacOS/macos-certificate-truster",
+            )
+        }
+        .expect("Failed to copy macos-certificate-truster. Has it been built yet?");
+
+        // macos-redirector app
+        let forlder_path = home_dir()
+            .unwrap()
+            .join("Library")
+            .join("Developer")
+            .join("Xcode")
+            .join("DerivedData");
+
+        let entries = fs::read_dir(forlder_path).unwrap();
+
+        //I need to do this because xcode renames the build folder with an ever-changing hash suffix,
+        //since previously I totally clean the DerivedData folder inside it there is only a name
+        //starting with MitmproxyAppleTunnel-
```

Comment:
```suggestion
        // xcode renames the build folder with an ever-changing hash suffix.
        // to work around this, we first clean DerivedData and then search for MitmproxyAppleTunnel-*
```

also, where are we doing this cleanup? only manually? (I vaguely recall seeing something somewere, but can't seem to find it)

Reply:
removed [here](https://github.com/mitmproxy/mitmproxy_rs/pull/66/commits/e1c185720019c0aa6ce5b588714db9775474dd9b), it was in the shell script called during CI 

I can add it again

Reply:
Ah, thanks! I take all the blame :)

I feel it does not really belong into the sign script, but we could create a build.sh script for local testing? If that's useful feel free to push it in a separate PR, no strong opinions. :)